File: csharp/DesignPattern.Interpreter/DivExpression.cs

namespace DesignPattern.Interpreter
{
    public class DivExpression: OperatorExpression
    {
        public DivExpression(Expression left, Expression right)
            : base(left, right)
        { 
        
        }

        public override double Interpret(Context context)
        {
            return this.left.Interpret(context) / this.right.Interpret(context);
        }
    }
}

File: c++/InterpreterPattern/DivExpression.cpp
#include "stdafx.h"
#include "DivExpression.h"


DivExpression::DivExpression(Expression *left,Expression *right) : OperatorExpression(left,right)
{

}


double DivExpression::Interpret(Context context)
{
	return this->left->Interpret(context) / this->right->Interpret(context);
}
